README.md

```diff
@@ -1,130 +1,214 @@
+3. After entering credentials the user will be presented with a list of their vehicles to authorize and a selection of permissions to grant your app.
+4. Once the user has selected their vehicle & selected permissions, this SDK will automagically trigger the `onComplete` function you provide with the authorization code or an error. If called with a code you must send this code to your backend to then exchange with Smartcar for an access token.
+
+    *Note:* Okay so about that automagically. Here's what happens behind the scenes. Let's say your website is `https://my-amazing-app.com`. Following the OAuth flow the user is redirected to the Smartcar-hosted redirect URI - `https://javascript-sdk.smartcar.com/redirect-2.0.0?app_origin=https://my-amazing-app.com&code={authorization-code}`. This page sources `redirect.js` which uses `postMessage` to send a message with the authorization code or error to the origin specified by the `app_origin` query param - `https://my-amazing-app.com`. The redirect page then closes itself out. On `https://my-amazing-app.com` the Smartcar instance you created receives the posted message and fires your `onComplete` method with the appropriate arguments. From your user's perspective all they'll see is a brief flicker of the redirect page before it closes out depending on their connection speed.
 5. Your application's backend server will need to accept the authorization code and exchange it for an access token.
 
-The SDK will help facilitate the OAuth link generation, popup dialog creation, and Smartcar will handle the user authentication and authorization. This SDK will not assist with the backend server code to accept authorization codes or exchanging for access tokens (step 6).
+For a more detailed explanation of our Smartcar-hosted redirect scheme see our [blog post](TODO: LINK HERE) on the topic.
+
+#### Self-Hosted
+
+Here's an example flow should you host the redirect yourself (many of the steps remain similar):
+
+1. User clicks "Connect your car" button (or similar) on your application's website.
+2. User is redirected to an OAuth authentication page, using the `openDialog`, `addClickHandler` or `generateLink` methods. This page requires the user to authenticate with their vehicle credentials.
+3. After entering credentials the user will be presented with a list of their vehicles to authorize and a selection of permissions to allow your app.
+4. Smartcar's services will redirect back to your application's redirect URI. The redirect URI will include a `code` query parameter holding the authorization code if the user approved the authorization. If they denied access the URI will include `error` & `error_description` parameters. If you source `redirect.js` on your redirect page it will trigger execution of your optionally provided `onComplete` method (passed to the `Smartcar` constructor on the client) and close out the redirect page.
```

Comment:
**If the user approves the authorization,** the redirect URI will include a `code` query parameter holding the authorization code. If they **deny access,** the URI will include `error` **and** `error_description` parameters. If you source `redirect.js` on your redirect **page,** it will